chia/full_node/full_node_api.py

```diff
@@ -1720,5 +1712,201 @@ def get_fee_estimates(est: FeeEstimatorInterface, req_times: List[uint64]) -> Li
         msg = make_msg(ProtocolMessageTypes.respond_fee_estimates, response)
         return msg
 
+    @api_request(
+        peer_required=True,
+        reply_types=[ProtocolMessageTypes.respond_add_puzzle_subscriptions],
+    )
+    async def request_add_puzzle_subscriptions(
+        self, request: wallet_protocol.RequestAddPuzzleSubscriptions, peer: WSChiaConnection
+    ) -> Message:
+        peer_id = peer.peer_node_id
+        max_subscriptions = self.max_subscriptions(peer)
+        subs = self.full_node.subscriptions
+
+        peak_height = self.full_node.blockchain.get_peak_height()
+        assert peak_height is not None
+
+        header_hash = self.full_node.blockchain.height_to_hash(peak_height)
+        assert header_hash is not None
```

Comment:
it's not obvious why it shouldn't be possible to subscribe to a node that hasn't synced yet. You could just return the genesis challenge as the previous block hash.
I suppose the peak height would be a bit tricky. But I believe you suggested it could be an `Optional[uint32]`

Reply:
Yeah I haven't been perfectly happy with this part either, I'll take a look at it again

Reply:
I've removed these messages.